src/go/rpk/pkg/tuners/executors/commands/create_sized_file.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// +build linux
+
+package commands
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+type writeSizedFileCommand struct {
+	path      string
+	sizeBytes int64
+}
+
+func NewWriteSizedFileCmd(path string, sizeBytes int64) Command {
+	return &writeSizedFileCommand{path, sizeBytes}
+}
+
+func (c *writeSizedFileCommand) Execute() error {
+	log.Debugf("Creating  '%s' (%d B)", c.path, c.sizeBytes)
+
+	_, err := os.Stat(c.path)
+	if err == nil {
+		// If the file exists, error out to prevent overwriting important files
+		// or filesystems.
+		return errBallastExists(c.path)
+	}
+	if !os.IsNotExist(err) {
+		return err
+	}
+
+	// the 'os' package needs to be used instead of 'afero', because the file
+	// handles returned by afero don't have a way to get their file descriptor
+	// (i.e. an Fd() method):
+	// https://github.com/spf13/afero/issues/234
+	//
+	// os.O_EXCL fails if O_CREATE and the file already exists, which is an
+	// additional safety net to prevent mistakes like overwriting an
+	// important file or even a whole filesystem.
+	f, err := os.OpenFile(c.path, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0666)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if err := unix.Fallocate(int(f.Fd()), 0, 0, c.sizeBytes); err != nil {
```

Comment:
i don't see you checking for size missmatch, .... ?

Reply:
How do you mean?

Reply:
if the ballast file size is different from run to run, you need to adjust. otherwise, it would be wrong.

Reply:
I think that only applies to if we allow resizing the ballast, which this PR does not currently allow?

Reply:
Fixed.